src/components/Form/FormProvider.js

```diff
@@ -260,6 +260,22 @@ const FormProvider = forwardRef(
                     inputValues[inputID] = _.isUndefined(propsToParse.defaultValue) ? getInitialValueByType(propsToParse.valueType) : propsToParse.defaultValue;
                 }
 
+                // If the input is a submit editing input, we need to set the onSubmitEditing prop
+                // to the submit function of the form
+                const onSubmitEditingObject = shouldSubmitEdit
+                    ? {
+                          onSubmitEditing: (e) => {
+                              submit();
+                              if (!propsToParse.onSubmitEditing) {
+                                  return;
+                              }
+                              propsToParse.onSubmitEditing(e);
+                          },
+                      }
+                    : {};
+
+                const isMultiline = propsToParse.multiline || propsToParse.autoGrowHeight;
```

Comment:
Doesn't this logic duplicate [this one](https://github.com/Expensify/App/pull/34471/files#diff-abde879b2542bb8feeb124550730f95fc100a648cc25683a5340271b4a7dc5adR20) in any way?